packages/core/src/conflict-resolution.ts

```diff
@@ -174,31 +167,45 @@ export async function fetchLog(
   dispatcher: Dispatcher,
   cid: CID,
   stateLog: HistoryLog,
-  log: CID[] = [],
-): Promise<CID[]> {
-  if (await stateLog.includes(cid)) {
+  log: LogEntry[] = [],
+): Promise<LogEntry[]> {
+  if (stateLog.includes(cid)) {
     // already processed
     return [];
   }
   const commit = await dispatcher.retrieveCommit(cid);
-  if (commit == null) {
-    throw new Error(`No commit found for CID ${cid.toString()}`);
-  }
+  if (!commit) throw new Error(`No commit found for CID ${cid.toString()}`);
 
   let payload = commit;
-  if (StreamUtils.isSignedCommit(commit)) {
+  const isSignedCommit = StreamUtils.isSignedCommit(commit);
+  if (isSignedCommit) {
     payload = await dispatcher.retrieveCommit(commit.link);
-    if (payload == null) {
-      throw new Error(`No commit found for CID ${commit.link.toString()}`);
-    }
+    if (!payload) throw new Error(`No commit found for CID ${commit.link.toString()}`);
   }
   const prevCid: CID = payload.prev;
   if (!prevCid) {
     // this is a fake log
     return [];
   }
-  log.push(cid); // Should be unshift [O(N)], but push [O(1)] + reverse [O(N)] seem better
-  if (await stateLog.includes(prevCid)) {
+  let nextEntry: LogEntry;
+  if (isSignedCommit) {
+    nextEntry = {
+      type: CommitType.SIGNED,
+      cid: cid,
+    };
+  } else {
+    // If not signed, then anchor commit
+    const proof = await dispatcher.retrieveFromIPFS(cid);
+    const timestamp = proof.blockTimestamp;
+    nextEntry = {
+      type: CommitType.ANCHOR,
+      cid: cid,
+      timestamp: timestamp,
+    };
+  }
+  // Should be unshift [O(N)], but push [O(1)] + reverse [O(N)] seem better
+  log.push(nextEntry);
+  if (stateLog.includes(prevCid)) {
     // we found the connection to the canonical log
     return log.reverse();
   }
```

Comment:
Can't comment on the next line but include `timestamp` variable to the `fetchLog` function call.